### DISCUSSION TITLE
Running MQ on AKS we expose UI and was able to connect to it and test using port-forward. We use Istio to handle routing and TLS, and when trying to expose the UI via Istio we’re seeing “upstream connect error or disconnect/reset before headers. reset reason: connection termination”

### DISCUSSION THREAD

**Discussion by veshwas** [istio/istio#52045] [Q&A]
Running MQ on AKS we expose UI and was able to connect to it and test using port-forward. We use Istio to handle routing and TLS, and when trying to expose the UI via Istio we’re seeing “upstream connect error or disconnect/reset before headers. reset reason: connection termination”

**Comment by keithmattix** [comment]
Are you injecting sidecars in your application or your MQ instances? Do you have logs you can share?

**Comment by veshwas** [comment]
We have sidecar on MQ instances

**Comment by keithmattix** [comment]
Are your clients connecting through an istio ingress gateway? What's the request flow?

**Comment by veshwas** [comment]
We have an application "A" which needs HTTPS traffic and we made changes to the gateway to allow traffic as is by setting TLS to PASSTHROUGH and matching the port in Virtual Service configuration" now we have a single host and all the https traffic is being routed to application A, I understand since HTTPS traffic is set to passthrough, we cannot add routes, is there a better way to approach this? To split the traffic to appropriate backend, is there a possible way?
